src/sentry/static/sentry/app/views/organizationGroupDetails/groupDetails.tsx

```diff
@@ -165,8 +191,29 @@ class GroupDetails extends React.Component<Props, State> {
       });
       const [data] = await Promise.all([groupPromise, eventPromise]);
 
-      // TODO(billy): See if this is even in use and if not, can we just rip this out?
+      const projects = organization.projects;
+      const projectId = data.project.id;
+      const features = projects.find(proj => proj.id === projectId)?.features ?? [];
+      // Check for the reprocessing-v2 feature flag
+      const hasReprocessingV2Feature = features.includes('reprocessing-v2');
+      const reprocessingStatus = getGroupReprocessingStatus(data);
+      const {currentTab, baseUrl} = this.getCurrentRouteInfo(data);
+
       if (this.props.params.groupId !== data.id) {
+        if (hasReprocessingV2Feature) {
+          // Redirects to the Activities tab
+          if (
+            reprocessingStatus === ReprocessingStatus.FULLY_REPROCESSED &&
+            currentTab !== TAB.ACTIVITY
+          ) {
```

Comment:
Should these two ifs be merged together?

Reply:
I think it's okay to leave it as it is, as the feature flag will be removed soon